### PR TITLE
Extend Blueprints to permit baseImageURI to be optionally overridden for each Homeserver

### PR DESCRIPTION
**Pull request by michaelkaye**
Fixes matrix-org/complement#435

Includes an error handling change to be able to see what's going on when base images aren't available. Before the error handling change, the requests below would have both reported the error:
`{"message":"failed to create deployment: CreateDeployment: Deploy returned error Deploy: No images have been built for blueprint custom"}`

Demonstrating the actual change by making these requests:

`curl -XPOST -d'{"base_image_uri":"not_an_image", "blueprint": { "Name": "custom", "Homeservers": [ { "Name": "hs1", "Users": [], "Rooms": [] }]}}' http://localhost:54321/create`
`{"message":"failed to create deployment: CreateDeployment: Failed to construct blueprint: ConstructBlueprintIfNotExist(custom): failed to ConstructBlueprint: errors whilst constructing blueprint custom: [Error response from daemon: No such image: not_an_image:latest]"}`

`curl -XPOST -d'{"base_image_uri":"not_an_image", "blueprint": { "Name": "custom", "Homeservers": [ { "Name": "hs1", "Users": [], "Rooms": [], "BaseImageURI": "alsonotanimage" }]}}' http://localhost:54321/create`
`{"message":"failed to create deployment: CreateDeployment: Failed to construct blueprint: ConstructBlueprintIfNotExist(custom): failed to ConstructBlueprint: errors whilst constructing blueprint custom: [Error response from daemon: No such image: alsonotanimage:latest]"}`

Also ran the synapse tester against this complement branch and it seemed to work fine.

### CODE REVIEW

**Review by kegsay — APPROVED** [review]
LGTM assuming CI is happy! Thanks!

**Comment by michaelkaye** [comment]
> LGTM assuming CI is happy! Thanks!

I don't have commit access to this repo, so if you could poke the button now everything's green, that'd be great.

**Comment by michaelkaye** [comment]
Actually: do not merge for the moment:
```
{"message":"failed to create deployment: CreateDeployment: Failed to construct blueprint: ConstructBlueprintIfNotExist(custom): failed to ConstructBlueprint: errors whilst constructing blueprint custom: [homerunner.custom.hs1: failed to check server is up. timed out checking for homeserver to be up: timed out checking for homeserver to be up: inspect container 602cca810dd9d78942b995b9bd52f2a92eff4dcd8870220f929f776fbb3b1a06 =\u003e health: starting]"}
```

I'm getting this locally when trying to spin up some servers; need to investigate. It hasn't been something I've changed that's causing the error - but it was hidden previously as the error wasn't propagated up to the http interface before.

**Comment by michaelkaye** [comment]
> Actually: do not merge for the moment:
> 
> ```
> {"message":"failed to create deployment: CreateDeployment: Failed to construct blueprint: ConstructBlueprintIfNotExist(custom): failed to ConstructBlueprint: errors whilst constructing blueprint custom: [homerunner.custom.hs1: failed to check server is up. timed out checking for homeserver to be up: timed out checking for homeserver to be up: inspect container 602cca810dd9d78942b995b9bd52f2a92eff4dcd8870220f929f776fbb3b1a06 =\u003e health: starting]"}
> ```
> 
> I'm getting this locally when trying to spin up some servers; need to investigate. It hasn't been something I've changed that's causing the error - but it was hidden previously as the error wasn't propagated up to the http interface before.

Actually, i think this is fine and is just covered by setting the `HOMERUNNER_SPAWN_HS_TIMEOUT_SECS` variable for homerunner correctly.

**Pull request merged**
